tests/test_lcov.py

```diff
@@ -0,0 +1,290 @@
+# Licensed under the Apache License: http://www.apache.org/licenses/LICENSE-2.0
+# For details: https://github.com/nedbat/coveragepy/blob/master/NOTICE.txt
+
+"""Test LCOV-based summary reporting for coverage.py"""
+
+import coverage
+
+
+from tests.coveragetest import CoverageTest
+
+
+class lcovTest(CoverageTest):
+    """Tests of the LCOV reports from coverage.py"""
+
+    def create_initial_files(self):
+        """
+        Helper for tests that handles the common ceremony so the tests can
+        show the consequences of changes in the setup
+        """
+        self.make_file(
+            "main_file.py",
+            """\
+            #!/usr/bin/env python3
+
+            def cuboid_volume(l):
+                return (l*l*l)
+
+            def IsItTrue():
+                return True
+
+        """,
+        )
+
+        self.make_file(
+            "test_file.py",
+            """\
+            #!/usr/bin/env python3
+
+            from main_file import cuboid_volume
+            import unittest
+
+            class TestCuboid(unittest.TestCase):
+                def test_volume(self):
+                    self.assertAlmostEqual(cuboid_volume(2),8)
+                    self.assertAlmostEqual(cuboid_volume(1),1)
+                    self.assertAlmostEqual(cuboid_volume(0),0)
+                    self.assertAlmostEqual(cuboid_volume(5.5),166.375)
+
+        """,
+        )
+
+    def run_coverage(self):
+        """Run coverage.py on main_file.py, and create an LCOV report."""
+        cov = coverage.Coverage(source=["."])
+        self.start_import_stop(cov, "main_file")
+        return cov
+
+    def get_lcov_report_content(self):
+        """Return the content of the LCOV report for `module`."""
+        filename = "coverage.lcov"
+        line_list = []
+        with open(filename) as f:
+            for line in f:
+                line_list.append(line)
+            return line_list
```

Comment:
I'm not a fan of returning a list of lines here.  I think the tests will read better if this function returns the text as one string, and the literals are multi-line strings instead of lists of lines.